File: dev/20190415/k2works/fizzbuzz/src/main/java/org/hiroshima_arc/domain/type/FizzBuzzType03.java
package org.hiroshima_arc.domain.type;

import org.hiroshima_arc.domain.model.FizzBuzzValue;

public class FizzBuzzType03 extends IFizzBuzzType {
    @Override
    public FizzBuzzValue generate(int number) {
        boolean fizz = isFizz(number);
        boolean buzz = isBuzz(number);
        if (fizz && buzz) return new FizzBuzzValue(number,"FizzBuzz");
        return new FizzBuzzValue(number,Integer.toString(number));
    }
}

File: dev/20190902/k2works/app/FizzBuzz/Domain/FizzBuzzType03.cs
namespace FizzBuzz.Domain
{
    public class FizzBuzzType03: FizzBuzzType
    {
        public override IFizzBuzz Generate(int number)
        {
            var isFizz = number % 3 == 0;
            var isBuzz = number % 5 == 0;
            if (isFizz && isBuzz)
            {
                return new FizzBuzzValue(number,FIZZ_BUZZ);
            }

            return new FizzBuzzValue(number,number.ToString());
        }
    }
}